### PR TITLE
feat(console): control console content by tenant env tag

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Control console content by tenant env tag
- Hide plan & billing page when the tenant is a dev tenant
- Display all pro tags when the tenant is a dev tenant

### Updates
- Refactor the `ProTag` component: use the `isVisibleInProdEnv` prop to control the visibility of the pro tag, it allows us to override the visibility inside the pro tag component according to the tenant env tag
- Extract all tenant notificaton modals in to a `TenantNotificationContainer` so we can manage them in one place.

## Testing

Test locally.

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] unit tests~~
~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/console/src/components/ProTag/index.tsx [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
+
+    return isVisible;
+  }, [isDevTenant, isVisible]);
+
+  return shouldDisplay ? (
     <div className={classNames(styles.tag, className)}>
       <DynamicT forKey="upsell.pro_tag" />
     </div>
-  );
+  ) : null;
```
Comment (charIeszhao) [inline packages/console/src/components/ProTag/index.tsx:30]: This is confusing. The `isVisible` prop indicates the tag should be displayed, while in fact there are extra condition checks inside the component. As a result, the show/hide status can be opposite with the `isVisible` prop...

File: packages/console/src/containers/AppContent/TenantNotificationContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+function TenantNotificationContainer() {
+  const { currentTenant, isDevTenant } = useContext(TenantsContext);
+  const isTenantSuspended = currentTenant?.isSuspended;
+
+  // Todo @xiaoyijun remove isDevFeaturesEnabled when the dev tenant feature is ready
+  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+  if (!isCloud || isTenantSuspended || (isDevFeaturesEnabled && isDevTenant)) {
+    return null;
+  }
```
Comment (charIeszhao) [inline packages/console/src/containers/AppContent/TenantNotificationContainer/index.tsx:16]: This condition seems to be problematic.
If `isDevFeaturesEnabled` is false && `isDevTenant` is true, then we'll see the modals below.
But clearly you would like to hide it from the UI in such situation.

Comment (charIeszhao) [reply packages/console/src/containers/AppContent/TenantNotificationContainer/index.tsx:16]: Should remove `isDevFeaturesEnabled &&` from this if condition, IMO

File: packages/console/src/components/ProTag/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/components/ProTag/index.tsx]: Updated by renaming the props name to `isVisibleInProdEnv`.

File: packages/console/src/components/CreateConnectorForm/index.tsx [hunk 3]
```diff
@@ -155,7 +155,7 @@ function CreateConnectorForm({ onClose, isOpen: isFormOpen, type }: Props) {
           <>
             <div className={styles.standardLabel}>
               <DynamicT forKey="connectors.standard_connectors" />
-              {isStandardConnectorDisabled && <ProTag />}
+              <ProTag isVisibleInProdEnv={isStandardConnectorDisabled} />
```
Comment (wangsijie) [inline packages/console/src/components/CreateConnectorForm/index.tsx:158]: Consider changing the name to "ProdTenant"?

Comment (charIeszhao) [reply packages/console/src/components/CreateConnectorForm/index.tsx:158]: Yeah, tenant is better than env

File: packages/console/src/components/ProTag/index.tsx [hunk 4]
```diff
@@ -1,19 +1,40 @@
...
+    if (!isCloud) {
+      return false;
+    }
+
+    if (isDevFeaturesEnabled && isDevTenant) {
+      return true;
+    }
+
+    return isVisibleInProdEnv;
+  }, [isDevTenant, isVisibleInProdEnv]);
```
Comment (charIeszhao) [inline packages/console/src/components/ProTag/index.tsx:31]: `useMemo` is not necessary for simple value calculations

File: packages/console/src/components/ProTag/index.tsx [hunk 5]
```diff
@@ -1,19 +1,40 @@
...
+     */
+    if (!isCloud) {
+      return false;
+    }
+
+    if (isDevFeaturesEnabled && isDevTenant) {
+      return true;
+    }
+
+    return isVisibleInProdEnv;
```
Comment (charIeszhao) [inline packages/console/src/components/ProTag/index.tsx:30]:
```suggestion
   if (isDevFeaturesEnabled && isDevTenant) {
      return isVisibleInProdEnv;
    }
```

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/components/CreateConnectorForm/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/components/CreateConnectorForm/index.tsx]: Updated.

**Pull request merged**